cli/Valet/Brew.php

```diff
@@ -7,6 +7,8 @@
 
 class Brew
 {
+    const SUPPORTED_PHP_VERSIONS = ['php', 'php@7.2', 'php@7.1', 'php@7.0', 'php@5.6'];
```

Comment:
As far as I'm aware `php` is equal to `php@7.2` so you should be able to omit `php` here. It will also save you some unnecessary code further down and not automatically enable php 7.3 support once that is released.

Reply:
I think you’re right.  The impetus of the new naming is to make getting Php easier and the latest version by default. `brew install php` always gets the latest. 

The only complication is that we have to make sure we always support the latest version before brew updates this tag.

Reply:
There are a number of issues with this.  Why introduce a constant SUPPORTED_PHP_VERSIONS when there is a function to return the same array, with the same name, 10 lines below.  Why introduce a regular expression to correct the values when you could just use the correct values?

Reply:
@epreston You're right about the `supportedPhpVersions` method is my error. But that method returns an instance of `collect` and in other contexts it is used as a PHP array.

As for the name of the versions, in Brew the new versions of PHP are known by those names. So when verifying if the formula was installed in the `installed` method, I could not find it and therefore I went from that formula.

Reply:
As a side note see toArray on collect if you ever need it.

Was the installed method failing because it didn’t have the correct values? The correct ones you just added that installed gets from that function?

Reply:
@epreston I know the `toArray` method but in this case the method should be` all`. But I think that the other parties should not need to know the class `collect` necessarily.

Before the changes, when installing the new versions of php and uninstalling the old ones. The formulas installed were `php, php@7.1, php@7.0, php @ 5.6` therefore when `installed`method executing the` brew list | grep formula` with `php71` for example I could not find the PHP version and it jumped.

Reply:
Exactly, I’ve been commenting on the exact same thing.  Check my history.

See the last few commits 2.0.9. It wasn’t released. I imagine the unit tests weren’t passing and I have a different build system for my test machines so I had no confidence in modifying the tests.

The side effect of installed not returning the correct result is that php doesn’t restart on some configurations.

Reply:
I just want to add that respect your time and effort. I’ve learned about the test suit from reading these commits.

My issue is that all this can be resolved by adding a few entries to the ones in the existing array rather than commits from a number of people adding regular expressions everywhere the array / collection is used. It’s hard to see complicated machinery introduced for a simple issue.  Worse when it doesn’t have backward compatibility when it’s easy to achieve, and more so when it’s not performant and clear.

Maybe I’m missing something in all this. Thank you for reading through and checking out the history on this.